src/WorkbenchExtension.ts

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+
+export class WorkbenchExtension {
+  // tslint:disable-next-line: no-any
+  private static extension: vscode.Extension<any>|undefined;
+
+  private constructor(context: vscode.ExtensionContext) {
+    // Get extensionId from package.json
+    const packageJsonPath = context.asAbsolutePath('./package.json');
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
```

Comment:
Suggest moving these part to a separate function, e.g. getExtensionId().
No need a constructor here.

Reply:
Fixed.